sdk/python/kfp/components/_components.py

```diff
@@ -202,12 +204,21 @@ def _create_task_factory_from_component_spec(component_spec:ComponentSpec, compo
 
     def create_task_from_component_and_arguments(pythonic_arguments):
         #Converting the argument names and not passing None arguments
-        valid_argument_types = (str, int, float, bool, GraphInputArgument, TaskOutputArgument) #Hack for passed PipelineParams. TODO: Remove the hack once they're no longer passed here.
+        valid_argument_types = (str, int, float, bool, GraphInputArgument, TaskOutputArgument, PipelineParam) #Hack for passed PipelineParams. TODO: Remove the hack once they're no longer passed here.
         arguments = {
             pythonic_name_to_input_name[k]: (v if isinstance(v, valid_argument_types) else str(v))
             for k, v in pythonic_arguments.items()
             if v is not None
         }
+        for key in arguments:
+            if isinstance(arguments[key], PipelineParam):
+                if kfp.TYPE_CHECK:
```

Comment:
nit: move it outside the loop?

Reply:
The PipelineParam argument needs to be converted to string. If the type check is moved outside, the value will remain the PipelineParam instance, which might lead to failure in the component module.

Reply:
sg